config/config.go

```diff
@@ -119,53 +127,73 @@ type Nodepool struct {
-	GKENetwork                          string   `json:"gkeNetwork" yaml:"gkeNetwork"`
-	GKEProjectID                        string   `json:"gkeProjectID" yaml:"gkeProjectID"`
-	GKESubnetwork                       string   `json:"gkeSubnetwork" yaml:"gkeSubnetwork"`
-	HostnamePrefix                      string   `json:"hostnamePrefix" yaml:"hostnamePrefix"`
-	LinodeImage                         string   `json:"linodeImage" yaml:"linodeImage"`
-	LinodeRootPass                      string   `json:"linodeRootPass" yaml:"linodeRootPass"`
-	LinodeToken                         string   `json:"linodeToken" yaml:"linodeToken"`
-	MachineConfigName                   string   `json:"machineConfigName" yaml:"machineConfigName"`
-	Module                              string   `json:"module" yaml:"module"`
-	NetworkPlugin                       string   `json:"networkPlugin" yaml:"networkPlugin"`
-	NodeTemplateName                    string   `json:"nodeTemplateName" yaml:"nodeTemplateName"`
-	OSDiskSizeGB                        int64    `json:"osDiskSizeGB" yaml:"osDiskSizeGB"`
-	PrivateAccess                       bool     `json:"privateAccess" yaml:"privateAccess"`
-	ProviderVersion                     string   `json:"providerVersion" yaml:"providerVersion"`
-	PublicAccess                        bool     `json:"publicAccess" yaml:"publicAccess"`
-	Region                              string   `json:"region" yaml:"region"`
-	ResourceGroup                       string   `json:"resourceGroup" yaml:"resourceGroup"`
-	ResourceLocation                    string   `json:"resourceLocation" yaml:"resourceLocation"`
-	VMSize                              string   `json:"vmSize" yaml:"vmSize"`
+	Ami                                 string                  `json:"ami,omitempty" yaml:"ami,omitempty"`
+	AvailabilityZones                   []string                `json:"availabilityZones" yaml:"availabilityZones"`
+	AWSAccessKey                        string                  `json:"awsAccessKey" yaml:"awsAccessKey"`
+	AWSInstanceType                     string                  `json:"awsInstanceType" yaml:"awsInstanceType"`
+	AWSRootSize                         int64                   `json:"awsRootSize" yaml:"awsRootSize"`
+	AWSSecretKey                        string                  `json:"awsSecretKey" yaml:"awsSecretKey"`
+	AWSSecurityGroupNames               []string                `json:"awsSecurityGroupNames" yaml:"awsSecurityGroupNames"`
+	AWSSecurityGroups                   []string                `json:"awsSecurityGroups" yaml:"awsSecurityGroups"`
+	AWSSubnetID                         string                  `json:"awsSubnetID" yaml:"awsSubnetID"`
+	AWSSubnets                          []string                `json:"awsSubnets" yaml:"awsSubnets"`
+	AWSVpcID                            string                  `json:"awsVpcID" yaml:"awsVpcID"`
+	AWSZoneLetter                       string                  `json:"awsZoneLetter" yaml:"awsZoneLetter"`
```

Comment:
seeing you break this out in set/setProviders.go made me realize, this should be broken out as well. In the config, we should have `aws`, `azure`, `linode`, etc. provider structs. Similar to how we do it in rancher/rancher. That way, the main terraformConfig isn't bloated with each individual provider's variables. i.e. vsphere will be gross as there are a ton of unique variables for it. 

could you make a new task to track this? Probably too much to be included in this PR.

Reply:
Great idea. I'll see how much it is to break it down here. If it's too much, I'll create a new task, but if it's not that big of a deal, I'll just update it in this PR.

Reply:
Resolved